lib/sort-order-selector/index.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import appState from '../flux/app-state';
+import {
+  toggleSearchSortOrder,
+  setSearchSortType,
+} from '../state/settings/actions';
+import classNames from 'classnames';
+import ArrowDownIcon from '../icons/arrow-down';
+
+export class SortOrderSelector extends Component {
+  static displayName = 'SortOrderSelector';
+
+  static propTypes = {
+    query: PropTypes.string.isRequired,
+    sortType: PropTypes.string.isRequired,
+    sortReversed: PropTypes.bool.isRequired,
+    searchSortType: PropTypes.string.isRequired,
+    searchSortReversed: PropTypes.bool.isRequired,
+    toggleSortOrder: PropTypes.func.isRequired,
+    setSortType: PropTypes.func.isRequired,
+  };
+
+  componentDidUpdate(prevProps) {
+    const { sortType, sortReversed, searchSortReversed, query } = this.props;
+    /* If there's a query and there wasn't before
+       (i.e., the search sort component is being newly displayed),
+       set the search sort to match the global setting.
```

Comment:
Why do we have local and global search orders? Would it make sense to only have a single global sort order that changes when we click to change it?

That would simplify the amount of code and complexity we have to introduce to support both sorting systems